test/contracts/provider/setup_test.go

```diff
@@ -0,0 +1,263 @@
+
+	rhdUserUUID, _ := uuid.NewUUID()
+	message := &createUserRequest{
+		createUserData: createUserData{
+			createUserAttributes: createUserAttributes{
+				Bio:       "Contract testing user account",
+				Cluster:   userCluster,
+				Email:     fmt.Sprintf("%s@redhat.com", userName),
+				Username:  userName,
+				RhdUserID: rhdUserUUID.String(),
+			},
+			Type: "identities",
+		},
+	}
+
+	messageBytes, err := json.Marshal(message)
+	if err != nil {
+		log.Fatalf("createUser: Unable to marshal JSON object:\n%q", err)
+	}
+
+	request, err := http.NewRequest("POST", fmt.Sprintf("%s/api/users", providerBaseURL), bytes.NewBuffer(messageBytes))
+	if err != nil {
+		log.Fatalf("createUser: Unable to create HTTP request:\n%q", err)
+	}
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", authServiceAccountToken))
+
+	log.Println("Sending a request to create a user")
+	response, err := httpClient.Do(request)
+	if err != nil {
+		log.Fatalf("createUser: Unable to send HTTP request:\n%q", err)
+	}
+	defer response.Body.Close()
+
+	responseBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Fatalf("createUser: Unable to read HTTP response:\n%q", err)
+	}
+
+	if response.StatusCode != 200 {
+		if response.StatusCode == 409 { //user already exists
+			log.Printf("User %s already exists, getting user info.", userName)
+			response2, err := http.Get(fmt.Sprintf("%s/api/users?filter[username]=%s", providerBaseURL, userName))
+			if err != nil {
+				log.Fatalf("userExists: Unable to create HTTP request:\n%q", err)
+			}
+			defer response2.Body.Close()
+
+			responseBody, err := ioutil.ReadAll(response2.Body)
+			// log.Printf("User info:\n%s\n", responseBody)
+			if err != nil {
+				log.Fatalf("userExists: Error reading HTTP response:\n%q", err)
+			}
+			if response2.StatusCode != 200 {
+				log.Fatalf("userExists: Something went wrong: %s", responseBody)
+			}
+			var users Users
+			err = json.Unmarshal(responseBody, &users)
+			if err != nil {
+				log.Fatalf("userExists: Unable to unmarshal response body: %s", err)
+			}
+			var user = &User{
+				Data: users.Data[0],
+			}
+			log.Printf("User found with ID: %s", user.Data.ID)
+			return user
+		}
+		log.Fatalf("createUser: Something went wrong with reading response body: %s", responseBody)
+	}
+
+	var user User
+	err = json.Unmarshal(responseBody, &user)
+	if err != nil {
+		log.Fatalf("createUser: Unable to unmarshal response body: %s", err)
+	}
+	log.Printf("User created with ID: %s", user.Data.ID)
+	return &user
+}
+
+// ServiceAccountTokenRequest represents a request JSON body
+type ServiceAccountTokenRequest struct {
+	GrantType    string `json:"grant_type"`
+	ClientID     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+}
+
+// ServiceAccountTokenResponse represents a response JSON body
+type ServiceAccountTokenResponse struct {
```

Comment:
This can be used from generated client, or fabric8-auth-client https://github.com/fabric8-services/fabric8-auth-client/blob/master/auth/media_types.go#L376

Reply:
Switched to auth-client.